### ISSUE TITLE
Unittests are fragile

### ISSUE THREAD

**Issue by shyuep** [materialsvirtuallab/maml#383]
@chc273 @JiQi535 @w6ye The unittests as they are written are very fragile. Look at the recent runs. When using ISIS, the results fluctuate and the tests fail randomly. The LBGS optimization also sometimes go out of bounds. While I understand that some of these algorithms are numerical in nature, tests have to be written on model systems and data where you can be very certain of the outcome. Otherwise, they are not proper tests. Fix this asap.

**Comment by chc273** [comment]
@shyuep Thanks for the comments. I have checked the SIS part. The fluctuation was caused by the near-zero impact from several data features in the test data, and those features can be easily swapped without affecting the final results. I have hence removed those non-important features. The SIS part itself is deterministic. That being said, SIS is not used in our work. It will need some further work in real applications to test the robustness of the method.

As for the BOWSR optimization, it does have some random elements, but the final optimized values should not change.  The error from BFGS was because of an issue noticed also from scipy community. It happens because the initial guess is slightly out of the set bounds of minimization. For example, it could happen when the initial value is -0 and the bound of this variable is [0, 1]. I have fixed the initial guess in the code and given it some buffer w.r.t. the bounds.
